lib/fizzy/cxx20/bit.hpp

```diff
@@ -0,0 +1,26 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <type_traits>
+
+namespace fizzy
+{
+/// The non-constexpr implementation of C++20's std::bit_cast.
+/// See https://en.cppreference.com/w/cpp/numeric/bit_cast.
+template <class To, class From>
+[[nodiscard]] inline
+    typename std::enable_if_t<sizeof(To) == sizeof(From) && std::is_trivially_copyable_v<From> &&
+                                  std::is_trivially_copyable_v<To>,
+        To>
+    bit_cast(const From& src) noexcept
```

Comment:
Are you planning to use this outside of the tests?

Reply:
Yes.

Reply:
In C API and execute() implementation.